tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -290,7 +302,21 @@ static void unprovisioned_beacon(uint8_t uuid[16],
 	if (uuid_to_provision && memcmp(uuid, uuid_to_provision, 16)) {
 		return;
 	}
-	bt_mesh_provision_adv(uuid, 0, prov_addr, 0);
+
+	if (memcmp(uuid, prov_uuid, 16) == 0) {
+		ASSERT_FALSE_MSG((prov_started && (k_uptime_delta(&prov_started) > 100)),
+				 "Received PB-ADV beacon after provisioning started.");
+	}
```

Comment:
Move this to a generic function which then can be used for both, pb-gatt and pb-adv. Move `100` to another macro and add a comment that it is taken taken randomly but reasonably small. (We can calculate it though here, not sure if we need this now. But if test starts failing, we can do this).